front/src/modules/trainschedule/components/ManageTrainSchedule/helpers/buildPacedTrainException.ts

```diff
@@ -173,3 +175,126 @@ export default function updatePacedTrainException(
     ? replaceElementAtIndex(originalPacedTrain.exceptions, exceptionIndex, exception)
     : removeElementAtIndex(originalPacedTrain.exceptions, exceptionIndex);
 }
+
+/**
+ * This function is after updating a paced train when the user send the form.
+ * It checks if an exception change group can be removed.
+ * If the paced train matches an exception change group value and if it was the last one remaining,
+ * the exception is removed.
+ */
+export function checkChangeGroups(
+  originalExceptions: PacedTrainException[],
+  updatedPacedTrain: PacedTrain
```

Comment:
I think you can change the type of the PacedTrain, so we clearly understand that we are building the new exceptions array
```suggestion
  updatedPacedTrain: Omit<PacedTrain, 'exceptions'>,
```

Reply:
`updatedPacedTrain` is only here for comparison. In the first line of the function, we can clearly see that we build a new exception array by looping on the original ones.

I don't really like changing the type here because this implies either : 
1. Removing the exceptions from the `updatedPacedTrain` before passing it to this function so more boilerplate
2. Leaving `updatedPacedTrain` as it is (since typescript doesn't complain) but we pass a `PacedTrain` and use inside an `Omit<PacedTrain, 'exceptions'>` which I think is weird

Reply:
Done